src/connection/xmpp/handlers/headlineMessage.ts

```diff
@@ -8,6 +8,12 @@ export default class extends AbstractHandler {
       let fromJid = new JID(stanza.getAttribute('from'));
       let connection = this.account.getConnection();
       let myJid = connection.getJID();
+      let nickname: string = $(stanza).find('nick').text();
```

Comment:
Same code as in chatMessage. Stay DRY.

Reply:
How would I reuse code from the method of another class?

Reply:
Put the code in it's own class for example. Therefore all nickname related code would be at one place. I think it would be a good semantic to group the code, because every contact has a nickname and it can be read and updated. Even every account has a own contact.